One Click Coolify Deployment

HowieDuhzit opened the pull request:
This is my setup for world deployment I have been using to deploy worlds in seconds.

This allows for near single click deployment on self hosted coolify servers by pulling a docker image file built by the DockerFile.

Currently the compose file is building from a docker image i maintain that is just an up to date build of dev branch:
howieduhzit/hyperfy:latest

Compose Features:
All the ENV, Storage, and port setup is done automatically
ENVs not commonly edited are hardcoded for convenience.
Admin code and JWT are set at random
World persistence seperate from app persistence so the updates are clean without destroying worlds
Persistence includes changing world variable between worlds.

Ideally we should have a main/dev branch docker image built from the hyperfy docker account so we can deploy as `hyperfy-xyz/hyperfy:main` or  `hyperfy-xyz/hyperfy:dev`

With this dockerfile its a simple as running
`docker build -t hyperfy-xyz/hyperfy:main .`
`docker push hyperfy-xyz/hyperfy:main`
or
`docker build -t hyperfy-xyz/hyperfy:dev .`
`docker push hyperfy-xyz/hyperfy:dev`
on each git push update

This allows for easy server updates by simply restarting the server to pull the latest image.

I will demo this at the next Hyperfy.How event to show how easy this makes deployment.

I also updated the public index and added the favicon and basic Hyperfy placeholder meta information for cleaner basic link sharing.

HPrivakos commented:
> Ideally we should have a main/dev branch docker image built from the hyperfy docker account so we can deploy as hyperfy-xyz/hyperfy:main or hyperfy-xyz/hyperfy:dev

We do, it's hosted on github repo instead of docker repo though, so it's `ghcr.io/hyperfy-xyz/hyperfy`
You have:
- dev: `ghcr.io/hyperfy-xyz/hyperfy:dev`
- main: `ghcr.io/hyperfy-xyz/hyperfy:main`
and every tags (since the docker auto build was setup)
- v0.8.1 for example: `ghcr.io/hyperfy-xyz/hyperfy:v0.8.1`
You can see the list here: [https://github.com/hyperfy-xyz/hyperfy/pkgs/container/hyperfy](https://github.com/hyperfy-xyz/hyperfy/pkgs/container/hyperfy)

I strongly disagree with the rollback to non-alpine node docker image, it unnecessarily make the image bigger while not adding any feature.

HowieDuhzit commented:
I fixed the dockerfile to be back in line with HP, updated the compose to use the main Image and also add a fix for the blackscreen issue with the app pane atm on dev branch

jb0gie commented:
![KevinGatesRbsIntroGIF](https://github.com/user-attachments/assets/8210cbf9-9018-4084-939c-1dd810bacbdb)
